src/main/java/randoop/sequence/Sequence.java

```diff
@@ -699,18 +699,30 @@ private Statement getStatementWithInputs(int index) {
    */
   public List<Variable> allVariablesForTypeLastStatement(Type type, boolean onlyReceivers) {
     List<Variable> possibleVars = new ArrayList<>(this.lastStatementVariables.size());
-    for (Variable i : this.lastStatementVariables) {
-      Statement s = statements.get(i.index);
-      Type outputType = s.getOutputType();
-      if (type.isAssignableFrom(outputType)
-          && !(onlyReceivers && outputType.isNonreceiverType())
-          && !(onlyReceivers && getCreatingStatement(i).isNonreceivingInitialization())) {
-        possibleVars.add(i);
+    for (Variable var : this.lastStatementVariables) {
+      if (matchesVariable(var, type, onlyReceivers)) {
+        possibleVars.add(var);
       }
     }
     return possibleVars;
   }
 
+  /**
```

Comment:
This is a reasonable change, but could you make a separate PR for it?  (You don't have to remove it from this PR; that will happen automatically when the other one is merged and master is merged into this one.)